support/run_template.html

```diff
@@ -23,19 +23,38 @@
 }
 
 
-function builtinRead(x)
-{
-    if (Sk.builtinFiles === undefined || Sk.builtinFiles["files"][x] === undefined)
+function builtinRead(x) {
+    console.log(x);
+    if (Sk.builtinFiles === undefined || Sk.builtinFiles["files"][x] === undefined) {
         throw "File not found: '" + x + "'";
-    return Sk.builtinFiles["files"][x];
+    }
+
+    var file = Sk.builtinFiles["files"][x];
+
+    if (!file) {
+        return new Promise(function (accept, reject) {
+            function reqListener () {
+                console.log(this);
+                console.log(arguments)
+                accept(this.responseText);
+            }
+
+            var oReq = new XMLHttpRequest();
+            oReq.addEventListener("load", reqListener);
+            oReq.open("GET", "%(root)s" + file);
+            oReq.send();
+        });
+    }
+
+    return file;
 }
 
 function runit(myDiv) {
    var prog = document.getElementById(myDiv+"_code").value;
    var mypre = document.getElementById(myDiv+"_pre");
 
-   Sk.inputfun = function() {
-       return new Promise(function (resolve) { resolve(window.prompt()); });
+   Sk.inputfun = function(prompt) {
+       return new Promise(function (resolve) { resolve(window.prompt(prompt)); });
```

Comment:
I think this  same pattern be applied to env.js where the base version of Sk.inputfun is defined?

It seems to work ok without the promise wrapper but I feel like it would be more correct.